compiler/p/codegen/OMRInstOpCodeProperties.hpp

```diff
@@ -11890,8 +12963,9 @@
    /* .mnemonic    = */ OMR::InstOpCode::paddi,
    /* .name        = */ "paddi",
    /* .description =    "Prefixed Add Immediate", */
-   /* .opcode      = */ 0x00000000,
-   /* .format      = */ FORMAT_UNKNOWN,
+   /* .prefix      = */ 0x06000000,
```

Comment:
I'm not entirely sure what you mean. This `prefix` field operates in exactly the same way as the old `opcode` field did, including the fact that its value is copied for record-form instructions. Can you please clarify?

Reply:
.opcode = metadata[addi].opcode, and
.propereties = metadata[addi].properties;

for type 2 & 3 prefixed instructions (they are modification to existing instructions).

Reply:
Ah, I see what you mean. I got confused and thought you were referring to something about the `prefix` field itself, as that's the line you were commenting on. My bad.

I don't think that would be a good idea IMO, as it would only apply to some prefixed instructions but not others. This could lead to considerable confusion for certain opcodes (e.g. `pld`/`pstd`) which operate as if they were modified versions of other instructions but which actually have unrelated encodings. The difference between these cases and other instructions where this does apply (e.g. `plwz`/`pstw`) is subtle and easily missed. In fact, AFAICT the ISA document only briefly mentions that these prefix types modify existing instructions in section 1.6.3, but itself proceeds to treat them in the same manner as other prefixed instructions in the remainder of the document; in the instruction descriptions, for example, the encodings of the prefixed forms of these instructions is listed separately from the encodings of their non-prefixed counterparts, making them look as if they're completely unrelated.

In contrast, the record-form variants of opcodes act in a very consistent manner. Either the instruction is record-form because it's implicitly so (there is no non-record-form equivalent at all) or it is identical to its non-record-form counterpart aside from modifying a condition register field, with a single bit differentiating them. Additionally, the ISA document generally does not provide separate encoding descriptions for the record-form variant of an instruction, instead adding an `Rc` field containing the bit to enable record-form, and even treats them as if they're the same instruction in the appendices that list all instructions. The only exception to this AFAIK is the `addic.` instruction, which uses a different primary opcode to `addic`.

I would consider the fact that type 2 & 3 prefixed instructions modify another instruction to be an implementation detail in hardware and I don't think we'd gain much by making use of it. IMO it's better to simply treat prefixed instructions as if they're completely unrelated at an encoding level to their non-prefixed versions.